utility/src/main/java/org/oppia/android/util/math/MathTokenizer.kt

```diff
@@ -0,0 +1,308 @@
+package org.oppia.android.util.math
+
+import java.lang.IllegalStateException
+import java.util.Locale
+
+// Only consider standard horizontal/vertical whitespace.
+private val VALID_WHITESPACE = listOf(' ', '\t', '\n', '\r')
+private val VALID_OPERATORS = listOf('*', '-', '+', '/', '^')
+private val VALID_DIGITS = listOf('0', '1', '2', '3', '4', '5', '6', '7', '8', '9')
+private val VALID_IDENTIFIER_LETTERS = "abcdefghijklmnopqrstuvwxyz".toCharArray().toList()
```

Comment:
According to me we only need to handle variables which are being added from the creator's end

Reply:
Ah. Are those customization args in the interactions? Can they be more than 1 letter long & can they be non-English letters?

Reply:
Yes they are customization arg and yes there non English letters like alpha, beta, gamma.
I don't think so that they are more than 1 letter long doesn't make any sense to me.

Reply:
Sorry to clarify, are you saying that they can be more than 1 letter long, or that they can't? If we used the greek letter alpha then that would still be 1 letter long, but if we use the English word "alpha" to denote 1 variable then the length is longer than 1.

Reply:
This is no longer relevant since the new PRD for the feature has clarified exactly how variables should behave, and the technical specification includes a formal CFG to define the syntax and format for variables.